tests/contrib/unittest_plugin/test_unittest.py

```diff
@@ -0,0 +1,361 @@
+import sys
+import unittest
+
+import pytest
+
+from ddtrace.constants import ERROR_MSG
+from ddtrace.constants import ERROR_TYPE
+from ddtrace.constants import SPAN_KIND
+from ddtrace.contrib.unittest.constants import COMPONENT_VALUE
+from ddtrace.contrib.unittest.constants import FRAMEWORK
+from ddtrace.contrib.unittest.constants import KIND
+from ddtrace.contrib.unittest.patch import _set_tracer
+from ddtrace.contrib.unittest.patch import patch
+from ddtrace.ext import SpanTypes
+from ddtrace.ext import test
+from ddtrace.internal.constants import COMPONENT
+from tests.utils import TracerTestCase
+from tests.utils import override_env
+
+
+class UnittestTestCase(TracerTestCase):
+    @pytest.fixture(autouse=True)
+    def fixtures(self, testdir, monkeypatch, git_repo):
+        self.testdir = testdir
+        self.monkeypatch = monkeypatch
+        self.git_repo = git_repo
+
+    with override_env(dict(DD_API_KEY="foobar.baz")):
+        patch()
+
+        def test_unittest_pass_single(self):
+            """Test with a `unittest` test which should pass."""
+            _set_tracer(self.tracer)
+
+            class UnittestExampleTestCase(unittest.TestCase):
+                def test_will_pass_first(self):
+                    self.assertTrue(2 == 2)
+                    self.assertTrue("test string" == "test string")
+                    self.assertFalse("not equal to" == "this")
+
+            suite = unittest.TestLoader().loadTestsFromTestCase(UnittestExampleTestCase)
+            unittest.TextTestRunner(verbosity=0).run(suite)
+
+            spans = self.pop_spans()
+            assert len(spans) == 1
+
+            assert spans[0].get_tag(test.TEST_TYPE) == SpanTypes.TEST
+            assert spans[0].get_tag(test.TEST_FRAMEWORK) == FRAMEWORK
+            assert spans[0].get_tag(SPAN_KIND) == KIND
+            assert spans[0].get_tag(COMPONENT) == COMPONENT_VALUE
+            assert spans[0].get_tag(test.TYPE) == SpanTypes.TEST
+            assert spans[0].get_tag(test.NAME) == "test_will_pass_first"
+            assert spans[0].get_tag(test.MODULE) == "tests.contrib.unittest_plugin.test_unittest"
+            assert spans[0].get_tag(test.SUITE) == "UnittestExampleTestCase"
+            assert spans[0].get_tag(test.TEST_STATUS) == test.Status.PASS.value
+
+        def test_unittest_pass_multiple(self):
+            """Tests with`unittest` tests which should pass."""
+            _set_tracer(self.tracer)
+
+            class UnittestExampleTestCase(unittest.TestCase):
```

Comment:
For a follow-up task/PR:

Does `unittest.TestLoader()` let you load test cases from multiple test classes? If so, I think it'd be worth writing at least one test where you have multiple test case classes, so we can make sure we're not accidentally overwriting the suite.

It might also be worth having one test where we do what we do with `pytest` and actually create modules on disk and call `unittest` on them.

Reply:
Good catch! I just added a test for this specific setup.